.github/workflows/node-ci-main.yml

```diff
@@ -7,6 +7,14 @@ on:
             - changeset-release/*
             - feature/*
 
+# When a PR is landed to main, we want to ensure that all of these jobs run and
+# that they all complete before the next PR's changes are processed by this
+# workflow.
+# See https://docs.github.com/en/actions/using-jobs/using-concurrency
+concurrency:
+    group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```

Comment:
I _think_ we could just use `gitHub.ref` for the second part of the group. I don't think we ever have `main` as the PR branch.

Reply:
Should the group just be something like `node-ci-main`? All CI runs for `main` could be in the same group, right?

Not that it should really matter, since cancel-in-progress is false.

Reply:
Actually, I think having the group just be `node-ci-main` is a good idea.  Although we have `cancel-in-progress: false`, we are saying that we are concurrent on the workflow (`node-ci-main`) and the PR number or Ref.  

So if multiple PRs land close together on main, they'd have different PR #s and/or refs which isn't what we want.  Thanks for calling this out!